.github/workflows/ci.yml

```
@@ -0,0 +1,59 @@
+      # Install dependencies
+      - run: npm install
+
+      # Run tests
+      - run: npm run test -- --coverage  # Assuming you're using Jest for testing
+
+      # Build the NestJS application (compiles TypeScript)
+      - run: npm run build
+
+      # Setup Node.js cache
+      - name: Cache dependencies
+        uses: actions/cache@v3
```

actionlint (1.7.4)

the runner of "actions/cache@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)